File: src/main/java/net/stefanfuchs/jslt/intellij/language/JsltSyntaxHighlighterFactory.java
package net.stefanfuchs.jslt.intellij.language;


import com.intellij.openapi.fileTypes.SyntaxHighlighterFactory;
import com.intellij.openapi.project.Project;
import com.intellij.openapi.vfs.VirtualFile;
import org.jetbrains.annotations.NotNull;

public class JsltSyntaxHighlighterFactory extends SyntaxHighlighterFactory {

  @NotNull
  @Override
  public JsltSyntaxHighlighter getSyntaxHighlighter(Project project, VirtualFile virtualFile) {
    return new JsltSyntaxHighlighter();
  }

}

File: src/main/kotlin/net/stefanfuchs/jslt/intellij/language/JsltSyntaxHighlighterFactory.kt
package net.stefanfuchs.jslt.intellij.language

import com.intellij.openapi.fileTypes.SyntaxHighlighterFactory
import com.intellij.openapi.project.Project
import com.intellij.openapi.vfs.VirtualFile

class JsltSyntaxHighlighterFactory : SyntaxHighlighterFactory() {
    override fun getSyntaxHighlighter(project: Project?, virtualFile: VirtualFile?): JsltSyntaxHighlighter {
        return JsltSyntaxHighlighter()
    }
}